tests/test_group.rs

```diff
@@ -312,106 +312,344 @@ fn group_operations() {
+        );
+        let (mls_plaintext_commit, _, kpb_option) = match group_bob.create_commit(
+            &[],
+            &bob_credential_bundle,
+            vec![update_proposal_bob.clone()],
+            false, /* force self update*/
+        ) {
+            Ok(c) => c,
+            Err(e) => panic!("Error creating commit: {:?}", e),
+        };
 
-    group_alice_1234.process_commit(commit5.clone());
-    group_bob.process_commit(commit5);
+        // Check that there is a new KeyPackageBundle
+        assert!(kpb_option.is_some());
+
+        group_alice
+            .apply_commit(
+                mls_plaintext_commit.clone(),
+                vec![update_proposal_bob.clone()],
+                vec![],
+            )
+            .expect("Error applying commit (Alice)");
+        group_bob
+            .apply_commit(
+                mls_plaintext_commit.clone(),
+                vec![update_proposal_bob],
+                vec![kpb_option.unwrap()],
+            )
+            .expect("Error applying commit (Bob)");
 
-    group_alice_1234.tree.print(&format!("\n{:?}", ms5));
+        // Make sure that both groups have the same public tree
+        if group_alice.tree().public_key_tree() != group_bob.tree().public_key_tree() {
+            _print_tree(&group_alice.tree(), "Alice added Bob");
+            panic!("Different public trees");
+        }
 
-    // Bob adds Charlie
-    let add_proposal = group_bob.create_add_proposal(&charlie_key_package, None);
-    group_alice_1234.process_proposal(add_proposal);
+        // === Alice updates and commits ===
+        let alice_update_key_package_bundle = KeyPackageBundle::new(
+            &[ciphersuite_name],
+            &alice_credential_bundle,
+            mandatory_extensions.clone(),
+        );
 
-    let (commit6, ms6, welcome_bundle_bob_charlie) = group_bob.create_commit(None);
+        let update_proposal_alice = group_alice.create_update_proposal(
+            &[],
+            &alice_credential_bundle,
+            alice_update_key_package_bundle.get_key_package().clone(),
+        );
+        let (mls_plaintext_commit, _, kpb_option) = match group_alice.create_commit(
+            &[],
+            &alice_credential_bundle,
+            vec![update_proposal_alice.clone()],
+            false, /* force self update*/
+        ) {
+            Ok(c) => c,
+            Err(e) => panic!("Error creating commit: {:?}", e),
+        };
 
-    group_alice_1234.process_commit(commit6.clone());
-    group_bob.process_commit(commit6);
+        // Check that there is a new KeyPackageBundle
+        assert!(kpb_option.is_some());
+
+        group_alice
+            .apply_commit(
+                mls_plaintext_commit.clone(),
+                vec![update_proposal_alice.clone()],
+                vec![kpb_option.unwrap()],
+            )
+            .expect("Error applying commit (Alice)");
+        group_bob
+            .apply_commit(
+                mls_plaintext_commit.clone(),
```

Comment:
`clone` shouldn't be necessary here (if at all, see comment above). (Same below.)

Reply:
Weird that clippy didn't catch that one. In general the cloning is needed because of the group simulation.

Reply:
Clippy is probably confused that the variable is shadowed later on.